File: ejercicios hechos/Python/estructuras alternativas/act17.py
#Programa: act17.py
#Propósito: Realiza un programa que pida por teclado el resultado (dato entero) obtenido al lanzar un dado de seis caras y muestre por pantalla el número en letras (dato cadena) de la cara opuesta al resultado obtenido.
#Autor: Jose Manuel Serrano Palomo.
#Fecha: 17/10/2019
#
#Variables a usar:
# cara_dado será la cara que el usuario introduzca
# cara_opuesta será el dato cadena para la cara opuesta
#
#Algoritmo:
# LEER cara_dado
# si cara_dado es igual a 1 escribir "seis"
# si cara_dado es igual a 2 escribir "cinco"
# si cara_dado es igual a 3 escribir "cuatro"
# si cara_dado es igual a 4 escribir "tres"
# si cara_dado es igual a 5 escribir "dos" 
# si cara_dado es igual a 6 escribir "uno"
# si cara_dado es mayor que 6 o menor que uno fallo

print("calcular la potencia")
print("---------------------\n")

#Leemos los datos
cara_dado = int(input("Introduce el numero que ha sacado el dado del 1 al 6: "))
cara_opuesta = ["uno","dos","tres","cuatro","cinco","seis"]

#Realizamos las comprobaciones
if cara_dado < 1 or cara_dado > 6:
    print("ERROR: número incorrecto")

else:
    if cara_dado == 1:
        print(f"En la cara opuesta está el {cara_opuesta[5]}")
    
    else:
        if cara_dado == 2:
            print(f"En la cara opuesta está el {cara_opuesta[4]}")
    
        else:
            if cara_dado == 3:
                print(f"En la cara opuesta está el {cara_opuesta[3]}")
    
            else:
                if cara_dado == 4:
                    print(f"En la cara opuesta está el {cara_opuesta[2]}")
    
                else:
                    if cara_dado == 2:
                        print(f"En la cara opuesta está el {cara_opuesta[1]}")
    
                    else:
                        if cara_dado == 1:
                            print(f"En la cara opuesta está el {cara_opuesta[0]}")

File: ejercicios terminados/Java/estructuras alternativas/act17.java
/*Programa: act17.java
 * 
 *Propósito: Realiza un programa que pida por teclado el resultado (dato entero) obtenido al lanzar un dado de seis caras y muestre por pantalla el número en letras (dato cadena) de la cara opuesta al resultado obtenido.
 * 
 * Nota 1: En las caras opuestas de un dado de seis caras están los números: 1-6, 2-5 y 3-4.
 * Nota 2: Si el número del dado introducido es menor que 1 o mayor que 6, se mostrará el mensaje: “ERROR: número incorrecto.”.
 * 
 *Autor: Jose Manuel Serrano Palomo.
 * 
 *Fecha: 09/11/2019
 *
 *Análisis:
 * Leemos la cara de un dado y debemos mostrar el opuesto en letras
 *
 *Diseño:
 * 1. Leer número de la cara
 * 2. Según el valor introducido:
 * 3. Si es = 1 muestro "SEIS"
 * 4. Si es = 2 muestro "CINCO"
 * 5. Si es = 3 muestro "CUATO"
 * 6. Si es = 4 muestro "TRES"
 * 7. Si es = 5 muestro "DOS"
 * 8. Si es = 6 muestro "UNO"
 * 9. Si es otro número, mostrar mensaje de error
*/
import java.util.Scanner;
 
public class act17 {
    public static void main(String[] args) {
        System.out.println("Programa para saber cara opuesta de un dado");
        System.out.println("-------------------------------------------");
        Scanner s = new Scanner(System.in);
        int cara;
        System.out.print("Introduce el número de la cara: ");
        cara = s.nextInt();

        switch (cara) {
          case 1:
            System.out.println("SEIS");
            break;
          case 2:
            System.out.println("CINCO");
            break;
          case 3:
            System.out.println("CUATRO");
            break;
          case 4:
            System.out.println("TRES");
            break;
          case 5:
            System.out.println("DOS");
            break;
          case 6:
            System.out.println("UNO");
            break;
          default:
            System.out.println("Error: número incorrecto.");
        }
    }
}
